From 22061bcd9f535fd112526f0dc6d730aa793d67c7 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 23 Dec 2020 20:40:57 +0200
Subject: [PATCH] [SW-26739] gaudi: security: open direct msi req

The direct MSI request must be used when in single MSI mode, so open the
direct MSI register mmPCIE_CORE_MSI_REQ.

Change-Id: If0d89159ee5d7da0d2e789e0d7de4fa048cdccc9
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index beb8bf6..ab9ebf0 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -12,6 +12,7 @@
 #include <cpu_if_regs.h>
 #include <psoc_global_conf_regs.h>
 #include <pcie_wrap_regs.h>
+#include <pcie_core_regs.h>
 #include <mme0_acc_regs.h>
 #include <mme0_sbab_regs.h>
 #include <mme0_ctrl_regs.h>
@@ -1125,41 +1126,47 @@
 	preboot_secure_lbw_rr(7, min, max);
 	preboot_secure_pcie_rr(7, min, max);
 
-	/* PCIE_CORE up to MMU_UP: RR_8 */
+	/* PCIE_CORE up to PCIE_CORE_L: RR_8 */
 	min = mmPCIE_CORE_BASE - 1;
-	max = mmMMU_UP_BASE;
+	max = HL_CFG_BASE + mmPCIE_CORE_MSI_REQ;
 	preboot_secure_lbw_rr(8, min, max);
 	preboot_secure_pcie_rr(8, min, max);
 
-	/* PSOC_I2C_M0 up to PSOC_TIMESTAMP: RR_9 */
-	min = mmPSOC_I2C_M0_BASE - 1;
-	max = mmPSOC_TIMESTAMP_BASE;
+	/* PCIE_CORE_H up to MMU_UP: RR_9 */
+	min = max + sizeof(u32) - 1;
+	max = mmMMU_UP_BASE;
 	preboot_secure_lbw_rr(9, min, max);
 	preboot_secure_pcie_rr(9, min, max);
 
-	/* PSOC_GPIO0 up to NIC0_MAC_CH0: RR_10 */
-	min = mmPSOC_GPIO0_BASE - 1;
-	max = mmNIC0_MAC_CH0_BASE;
+	/* PSOC_I2C_M0 up to PSOC_TIMESTAMP: RR_10 */
+	min = mmPSOC_I2C_M0_BASE - 1;
+	max = mmPSOC_TIMESTAMP_BASE;
 	preboot_secure_lbw_rr(10, min, max);
 	preboot_secure_pcie_rr(10, min, max);
 
-	/* NIC0_TS up to NIC1_MAC_CH0: RR_11 */
-	min = mmNIC0_TS_BASE - 1;
-	max = mmNIC1_MAC_CH0_BASE;
+	/* PSOC_GPIO0 up to NIC0_MAC_CH0: RR_11 */
+	min = mmPSOC_GPIO0_BASE - 1;
+	max = mmNIC0_MAC_CH0_BASE;
 	preboot_secure_lbw_rr(11, min, max);
 	preboot_secure_pcie_rr(11, min, max);
 
-	/* NIC1_TS up to NIC2_MAC_CH0: RR_12 */
-	min = mmNIC1_TS_BASE - 1;
-	max = mmNIC2_MAC_CH0_BASE;
+	/* NIC0_TS up to NIC1_MAC_CH0: RR_12 */
+	min = mmNIC0_TS_BASE - 1;
+	max = mmNIC1_MAC_CH0_BASE;
 	preboot_secure_lbw_rr(12, min, max);
 	preboot_secure_pcie_rr(12, min, max);
 
-	/* NIC3_TS up to NIC4_MAC_CH0: RR_13 */
-	min = mmNIC3_TS_BASE - 1;
-	max = mmNIC4_MAC_CH0_BASE;
+	/* NIC1_TS up to NIC2_MAC_CH0: RR_13 */
+	min = mmNIC1_TS_BASE - 1;
+	max = mmNIC2_MAC_CH0_BASE;
 	preboot_secure_lbw_rr(13, min, max);
 	preboot_secure_pcie_rr(13, min, max);
+
+	/* NIC3_TS up to NIC4_MAC_CH0: RR_14 */
+	min = mmNIC3_TS_BASE - 1;
+	max = mmNIC4_MAC_CH0_BASE;
+	preboot_secure_lbw_rr(14, min, max);
+	preboot_secure_pcie_rr(14, min, max);
 }
 
 static void preboot_secure_hbw_rr(int rr, u64 base, u64 mask)
